tests/run-make/core-ffi-typecheck-clang/rmake.rs

```diff
@@ -0,0 +1,210 @@
+//@ needs-force-clang-based-tests
+// This test checks that the clang defines for each target allign with the core ffi types defined in
+// mod.rs. Therefore each rust target is queried and the clang defines for each target are read and
+// compared to the core sizes to verify all types and sizes allign at buildtime.
+//
+// If this test fails because Rust adds a target that Clang does not support, this target should be
+// added to SKIPPED_TARGETS.
+
+use run_make_support::{clang, llvm_components_contain, regex, rfs, rustc};
+
+// It is not possible to run the Rust test-suite on these targets.
+const SKIPPED_TARGETS: &[&str] = &[
+    "wasm32v1-none",
+    "xtensa-esp32-espidf",
+    "xtensa-esp32-none-elf",
+    "xtensa-esp32s2-espidf",
+    "xtensa-esp32s2-none-elf",
+    "xtensa-esp32s3-espidf",
+    "xtensa-esp32s3-none-elf",
+];
+
+/// Map from a Rust target to the Clang target if they are not the same.
+const MAPPED_TARGETS: &[(&str, &str)] = &[
+    ("aarch64-apple-ios-sim", "aarch64-apple-ios"),
+    ("aarch64-apple-tvos-sim", "aarch64-apple-tvos"),
+    ("aarch64-apple-visionos-sim", "aarch64-apple-visionos"),
+    ("aarch64-apple-watchos-sim", "aarch64-apple-watchos"),
+    ("x86_64-apple-watchos-sim", "x86_64-apple-watchos"),
+];
+
+fn main() {
+    let targets = get_target_list();
+
+    let minicore_path = run_make_support::source_root().join("tests/auxiliary/minicore.rs");
+
+    regex_mod();
+
+    for target in targets.lines() {
+        if SKIPPED_TARGETS.iter().any(|&to_skip_target| target == to_skip_target) {
+            continue;
+        }
+
+        // Map the Rust target string to a Clang target string if needed
+        // Also replace riscv with necessary replacements to match clang
+        // If neither just use target string
+        let ctarget = MAPPED_TARGETS
+            .iter()
+            .find(|(rtarget, _)| *rtarget == target)
+            .map(|(_, ctarget)| ctarget.to_string())
+            .unwrap_or_else(|| {
+                if target.starts_with("riscv") {
+                    target
+                        .replace("imac", "")
+                        .replace("gc", "")
+                        .replace("imafc", "")
+                        .replace("imc", "")
+                        .replace("ima", "")
+                        .replace("im", "")
+                        .replace("emc", "")
+                        .replace("em", "")
+                        .replace("e-", "-")
+                        .replace("i-", "-")
```

Comment:
Just to reduce the chance of matching the middle of a future OS name or something

```suggestion
                    target
                        .replace("imac-", "-")
                        .replace("gc-", "-")
                        .replace("imafc-", "-")
                        .replace("imc-", "-")
                        .replace("ima-", "-")
                        .replace("im-", "-")
                        .replace("emc-", "-")
                        .replace("em-", "-")
                        .replace("e-", "-")
                        .replace("i-", "-")
```